src/menu.tsx

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from "react";
+
+import { AlertVariant } from "@patternfly/react-core/dist/esm/components/Alert";
+
+import cockpit from "cockpit";
+
+import type { FolderFileInfo } from "./app";
+import { basename } from "./common";
+import { CreateDirectoryModal } from "./dialogs/mkdir";
+import {
+    ConfirmDeletionDialog,
+    RenameItemModal,
+    editPermissions,
+    downloadFile
+} from "./fileActions";
+import type { FileInfo } from "./fsinfo";
+
+const _ = cockpit.gettext;
+
+type MenuItem = { type: "divider" } | {
+    type?: never,
```

Comment:
> TypeScript introduced a new type never, which indicates the values that will never occur.

But uh, this will be set to a string at some point?

Reply:
No.

This is basically a way of discriminating the union: either we have items that set `type` to the literal string `"divider"`, or it's forbidden from appearing at all (`never`).